File: JUNE-2025/28-06-2025/p2099.java
import java.util.PriorityQueue;

class Solution {
    public int[] maxSubsequence(int[] nums, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> Integer.compare(a[0], b[0]));
        for (int i = 0; i < nums.length; i++) {
            pq.offer(new int[] { nums[i], i });
            if (pq.size() > k)
                pq.poll();
        }

        PriorityQueue<int[]> newPq = new PriorityQueue<>((a, b) -> Integer.compare(a[1], b[1]));
        newPq.addAll(pq);

        int[] res = new int[k];
        int i = 0;

        while (!newPq.isEmpty()) {
            int[] val = newPq.poll();
            res[i++] = val[0];
        }

        return res;
    }
}

File: JUNE-2025/28-06-2025/p2099.kt
class Solution {
    fun maxSubsequence(nums: IntArray, k: Int): IntArray {
        val ascIndices = PriorityQueue<Int> { a, b ->
            nums[a] - nums[b]
        }
        for (i in nums.indices) {
            ascIndices.add(i)
            if (ascIndices.size > k) {
                ascIndices.poll()
            }
        }
        val result = IntArray(k)
        var idx = 0
        for (i in nums.indices) {
            if (ascIndices.contains(i)) {
                result[idx++] = nums[i]
            }
        }
        return result
    }
}